### ISSUE TITLE
Online DPO

### ISSUE THREAD

**Issue by qgallouedec** [huggingface/trl#2623]

**Comment by qgallouedec** [comment]
```python
from datasets import load_dataset
from transformers import AutoModelForCausalLM, AutoTokenizer, AutoModelForSequenceClassification
from trl import OnlineDPOConfig, OnlineDPOTrainer


model_id = "trl-internal-testing/tiny-Qwen2ForCausalLM-2.5"
model = AutoModelForCausalLM.from_pretrained(model_id)
ref_model = AutoModelForCausalLM.from_pretrained(model_id)
tokenizer = AutoTokenizer.from_pretrained(model_id)
tokenizer.pad_token = tokenizer.eos_token

reward_model = AutoModelForSequenceClassification.from_pretrained("trl-internal-testing/tiny-Qwen2ForSequenceClassification-2.5")
reward_tokenizer = AutoTokenizer.from_pretrained("trl-internal-testing/tiny-Qwen2ForSequenceClassification-2.5")

batch_size = 8
gradient_accumulation_steps = 1
output_dir = f"OnlineDPO-bsz{batch_size}-grad_acc{gradient_accumulation_steps}"

training_args = OnlineDPOConfig(
    output_dir=output_dir,
    per_device_train_batch_size=batch_size,
    gradient_accumulation_steps=gradient_accumulation_steps,
    logging_steps=2,
)

dummy_dataset = load_dataset("trl-internal-testing/zen", "standard_prompt_only")

trainer = OnlineDPOTrainer(
    model=model,
    ref_model=ref_model,
    reward_model=reward_model,
    args=training_args,
    processing_class=tokenizer,
    reward_processing_class=reward_tokenizer,
    train_dataset=dummy_dataset["train"],
)

previous_trainable_params = {n: param.clone() for n, param in trainer.model.named_parameters()}

trainer.train()
```


![Image](https://github.com/user-attachments/assets/3dcaec17-59c2-4651-94a2-f0368a44a295)

No issue here. It's because we handle the loss.backward:

https://github.com/huggingface/trl/blob/40c238395e345e6013f899b3768b53c73e60844b/trl/trainer/online_dpo_trainer.py#L693
